src/librustc_trans/back/rpath.rs

```diff
@@ -41,7 +41,7 @@ pub fn get_rpath_flags(config: &mut RPathConfig) -> Vec<String> {
     flags.extend_from_slice(&rpaths_to_flags(&rpaths));
 
     // Use DT_RUNPATH instead of DT_RPATH if available
-    if config.linker_is_gnu {
+    if config.linker_is_gnu && target.target_os == "linux" {
```

Comment:
There is no variable "target" in this function.

Reply:
Sorry, I would check where to get it from, but this breaks bootstrap for me so I'm typing "blind"